tests/ReactTestApp/metro.config.js

```diff
@@ -1,15 +1,162 @@
+  const combinedLinks = [...directLinks, ...scopedLinks];
+  const links = combinedLinks.map(({name, path: linkPath}) => {
+    const linkPackageJson = readJson(linkPath, 'package.json');
+    const peerDependencies = Object.keys(
+      linkPackageJson.peerDependencies || {},
+    );
+    return {name, path: linkPath, peerDependencies};
+  });
+  // We're only interested in actual dependencies
+  const dependencyLinks = links.filter(
+    ({name}) => name in packageJson.dependencies && !exclude.has(name),
+  );
+  // Recurse
+  // Ensure we don't visit this package again
+  exclude.add(packageJson.name);
+  // Traverse all dependencies
+  for (const link of dependencyLinks) {
+    const additionalLinks = getLinkedDependencies(link.path, exclude);
+    dependencyLinks.push(...additionalLinks);
+  }
+  // Return only new links
+  return dependencyLinks;
+}
+
+const linkedDependencies = getLinkedDependencies(__dirname);
+console.log(
+  `Linked dependencies: ${linkedDependencies.map(({name}) => name).join(', ')}`,
+);
+
+const watchFolders = linkedDependencies.map(pkg => pkg.path);
+
+const blockedPaths = [];
+for (const pkg of linkedDependencies) {
+  // Block the links
+  blockedPaths.push(path.join(__dirname, 'node_modules', pkg.name));
+  // Block any peer dependencies of the links
+  blockedPaths.push(
+    ...pkg.peerDependencies.map(peer =>
+      path.join(pkg.path, 'node_modules', peer),
+    ),
+  );
+}
+// Turn paths into regular expressions
+const blockList = blockedPaths.map(
+  pkgPath => new RegExp(`^${escape(pkgPath)}\\/.*$`),
+);
+
+const peerNodeModules = {};
+const allPeerDependencies = linkedDependencies.flatMap(
+  pkg => pkg.peerDependencies,
+);
+const unresolvedPeerDependencies = new Set(
+  allPeerDependencies.filter(name => !(name in linkedDependencies)),
+);
+// Locate the package providing every unresolved peer dependency
+for (const peer of unresolvedPeerDependencies) {
+  for (const pkg of linkedDependencies) {
+    const potentialPath = path.join(pkg.path, 'node_modules', peer);
+    if (fs.existsSync(potentialPath)) {
+      peerNodeModules[peer] = potentialPath;
+      break;
+    }
+  }
+}
+// Create a proxy around the resolved peer dependencies
+// This will resolve all modules exposed by the react-native app and fall back to resolved peer dependencies
+const extraNodeModules = new Proxy(peerNodeModules, {
+  get(target, name, receiver) {
+    const potentialPath = path.join(__dirname, 'node_modules', name);
+    if (fs.existsSync(potentialPath)) {
+      return potentialPath;
+    } else if (name in target) {
+      return target[name];
+    } else {
+      return undefined;
+    }
+  },
+});
 
 module.exports = {
+  projectRoot: __dirname,
+  watchFolders,
+  resolver: {
+    blockList: exclusionList(blockList),
+    extraNodeModules,
+  },
   transformer: {
     getTransformOptions: async () => ({
       transform: {
-        experimentalImportSupport: false,
         inlineRequires: true,
       },
     }),
```

Comment:
Could we reuse the [metro-config](https://github.com/realm/realm-js/tree/master/packages/metro-config) package instead of copying it here?

Reply:
Actually the github action isn't even using the testapp.  I did however make a PR where I attempt to add this:
https://github.com/realm/realm-js/pull/4368